Simsensor docs (#151)
phoenix-rtos/phoenix-rtos-doc: devices/sensors/simsensors.md

The concrete implementation of the Simsensor publishes prepared measurement readings from a special file.

## Supported sensors

Currently, all types of sensors supported by Sensorhub have their simulated type. The naming convention assumes the use of the `_sim` suffix, for example, `baro_sim` for the barometer.

## File format

The concrete simulated sensor needs a special `CSV` file, which contains measurements saved in table structured format.

### Columns

 - The first column contains the ID of a sensor type, which corresponds to IDs from Sensorhub. The first column contains the ID of a sensor type, which corresponds to the IDs from Sensorhub. Furthermore, as described in the `Measurement publishing algorithm` section, the end scenario indicator can be used as a value of this field. All currently available values are listed below.

| Sensor type   | ID |
|---------------|----|
| End scenario  |  0 |
| Accelerometer |  1 |
| Barometer     |  2 |
| GPS           |  4 |
| Gyroscope     |  8 |
| Magnetometer  | 16 |
| Thermometer   | 32 |

>  - The second column contains a timestamp of a particular measurement. The timestamp should be in **milliseconds**. It is important to note that it does not have a real-time timestamp. The precise meaning of a timestamp is described in the section `Measurement publishing algorithm`.

Reviewer: Did you mean
```suggestion
 - The second column contains a timestamp of a particular measurement. The timestamp should be in **milliseconds**. It is important to note that it does not have to be a real-time timestamp. The precise meaning of a timestamp is described in the section `Measurement publishing algorithm`.
```
?

reading all measurements. Timestamps monotonicity is preserved. Time span between the last reading and the first one
from file is considered to be equal to 0. In practice the time between publishes is as small as possible.

It is also possible to end the scenario. For this purpose special end scenario indicator - `0` should be placed in
first column (as described in [this section](#columns)). Simsensor doesn't publish any new events after this symbol.

### File example

```csv
SensorID,Timestamp,Field1,Field2,Field3
1,2282607175,748,-419,-9780,0
8,2282607175,-2,-64,0,0
1,2282610344,740,-484,-9776,0
8,2282610344,-6,-71,-2,0
1,2282612689,740,-484,-9776,0
8,2282612689,-6,-73,-3,0
1,2282614538,742,-510,-9788,0
8,2282614538,-1,-68,-2,0
1,2282617192,723,-517,-9783,0
```

## Usage

Running selected Simsensor is analogical to start collecting data from a physical device. There is only one required
argument: the path to a file containing the measurement scenario to be used. Command to start sensor should look file
this:

```bash
./sbin/sensors -s <sensor_type>_sim:<scenario_file_path>
```

> From perspective of data consumer, nothing is changing.

Reviewer: ```suggestion
From perspective of data consumer, nothing changes.
```

### Starting simsensor

Assume that mocking barometer data is needed and path to file with data to be published is

> `./etc/mock_measurements.csv`. Then the command to start looks like this:
> 
> ```bash
> ./sbin/sensors -s baro_sim:<scenario_file_path>
> ```

Reviewer: ```suggestion
Assume that mocking barometer data is needed and path to file with data to be published is
`./etc/mock_measurements.csv`. Then the command to start looks like this:

```bash
./sbin/sensors -s baro_sim:./etc/mock_measurements.csv
```
```
